### PR TITLE
Time space diagram

### PR DESCRIPTION
**Pull request by AboudyKreidieh**
prerequisite: PR #512 

changes:
- added a method for plotting time-space diagrams
- using the time-space diagrams to modify the vehicle parameters in the simulations to match expected behaviors in new sumo (more on this to come)
- tests for the new time-space diagram plotter

### CODE REVIEW

**Comment by AboudyKreidieh** [comment]
The figure eight time-space diagram originally looks like this:

Selection_048

In order to achieve a similar (one-for-one) behavior, I've adjusted the comfortable decel value in sumo to be 1.5 m/s2 for this setting. Note, however, that the curve are closer together. This may have an effect on training ultimately. More on this probably in a later PR.

Selection_025

**Comment by AboudyKreidieh** [comment]
The merge time-space diagrams in old sumo look like this:

Selection_048

**Comment by AboudyKreidieh** [comment]
When switching to new sumo, the car are more likely to collide. To fix this, while not affecting the performance of the network too greatly, I increased the "radius" term of the merge junction (i.e. "center") to 10m. The resulting behavior looks something like this:

Figure_3

**Comment by AboudyKreidieh** [comment]
The behavior looks better over larger time frames:

Figure_4

File: tests/fast_tests/test_files/fig8.json [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+        "sims_per_step": 1,
+        "warmup_steps": 0
+    },
+    "env_name": "AccelEnv",
+    "exp_tag": "figure_eight_0",
+    "initial": {
+        "additional_params": {},
+        "bunching": 0,
+        "edges_distribution": "all",
+        "lanes_distribution": Infinity,
```
Comment (Yasharzf) [inline tests/fast_tests/test_files/fig8.json]: Is this a syntax error?

File: tests/fast_tests/test_files/loop_230.json [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+        "sims_per_step": 1,
+        "warmup_steps": 750
+    },
+    "env_name": "WaveAttenuationPOEnv",
+    "exp_tag": "stabilizing_the_ring",
+    "initial": {
+        "additional_params": {},
+        "bunching": 0,
+        "edges_distribution": "all",
+        "lanes_distribution": Infinity,
```
Comment (Yasharzf) [inline tests/fast_tests/test_files/loop_230.json]: syntax error??

File: tests/fast_tests/test_files/merge.json [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+        "sims_per_step": 2,
+        "warmup_steps": 0
+    },
+    "env_name": "WaveAttenuationMergePOEnv",
+    "exp_tag": "merge_0",
+    "initial": {
+        "additional_params": {},
+        "bunching": 0,
+        "edges_distribution": "all",
+        "lanes_distribution": Infinity,
```
Comment (Yasharzf) [inline tests/fast_tests/test_files/merge.json]: ??

File: tests/fast_tests/test_files/fig8.json (hunk 1 above)
Comment (AboudyKreidieh) [reply tests/fast_tests/test_files/fig8.json]: Oh no, it's how json stores inf numbers. It doesn't cause any bug though, don't worry.

**Pull request merged**